interactions/api/models/guild.py

```diff
@@ -1638,14 +1638,21 @@ async def get_role(
         :rtype: Role
         """
 
-        if not self._client:
-            raise LibraryException(code=13)
-        roles = await self._client.get_all_roles(guild_id=int(self.id))
-        for i in roles:
-            if int(i["id"]) == role_id:
-                role = Role(**i)
-                break
-        return role
+        for role in self.roles:
+            if int(role.id) == role_id:
+                return role
+        else:
+            if not self._client:
+                raise LibraryException(code=13)
+            roles = await self._client.get_all_roles(guild_id=int(self.id))
+            self.roles = [Role(**_) for _ in roles]
+            for role in self.roles:
+                if int(role.id) == role_id:
+                    return role
+            else:
+                raise LibraryException(
+                    message="The role you looked for was not found!", code=0, severity=30
+                )
```

Comment:
Tbh you don't need the else bits since it's a return anyways

```suggestion
        for role in self.roles:
            if int(role.id) == role_id:
                return role
        if not self._client:
            raise LibraryException(code=13)
        roles = await self._client.get_all_roles(guild_id=int(self.id))
        self.roles = [Role(**_) for _ in roles]
        for role in self.roles:
            if int(role.id) == role_id:
                return role
        raise LibraryException(
            message="The role you looked for was not found!", code=0, severity=30
        )
```
This would make it a little less readable though, lmk what you think

Reply:
It's OK ig